src/array.jl

```diff
@@ -642,19 +656,48 @@ function Base.reinterpret(::Type{T}, a::CuArray{S,N}) where {T,S,N}
     osize = tuple(size1, Base.tail(isize)...)
   end
 
-  refcount = a.storage.refcount[]
-  @assert refcount != 0
-  if refcount > 0
-    Threads.atomic_add!(a.storage.refcount, 1)
-  end
+  return _create_reinterpret(T, N, a, osize)
+end
 
-  b = CuArray{T,N}(a.storage, osize; a.maxsize, a.offset)
-  if refcount > 0
-      finalizer(unsafe_finalize!, b)
+# modified from Base: reinterpretarray.jl:14-77
```

Comment:
Too bad we need to duplicate all this.

Reply:
The definitions in Base are defined inside the constructor of `ReinterpretedArray`, maybe I should also submit a PR to clean up this a bit in Base so that we don't need to duplicate this in the future. For the already released julia versions this would be necessary I think